### PR TITLE
Add azure devops repo init

### PR DESCRIPTION
**Pull request by drosevear**
This PR add a `.azuredevops` directory that accelerates an MLOps project template implementation in Azure DevOps. More particularly, it:

- Allows users to run a pipeline that creates a project structure according to their selection (equivalent to the 'sparse checkout') in an empty repo.
- Creates new Azure DevOps pipelines for infrastructure and MLOps based on the Azure Pipelines yaml files in `infrastructure/pipelines` and `mlops/devops-pipelines` respectively.

### CODE REVIEW

**Comment by kevball2** [comment]
@cindyweng  was that name change something I could updated on the PR. I didn't think to try!

**Comment by setuc** [comment]
@drosevear Thank you very much for your contribution. We can fill this with the documentation for folks who want to get started with Azure DevOps as their code repository.

**Comment by kevball2** [comment]
@setuc is specific documentation you would like to see for this new functionality? I would like to test this functionality with AzDo Repositories and would be willing to write the required documentation.

**Comment by setuc** [comment]
@dhangerkapil  for your review

**Comment by setuc** [comment]
> @setuc is specific documentation you would like to see for this new functionality? I would like to test this functionality with AzDo Repositories and would be willing to write the required documentation.

@kevball2  Apologies for the delay, somehow I am not getting any alerts on the comments. Anyways, the steps that we are looking for is someone who is starting off, how they can make use of these steps in ADO, potentially with a screenshot. Thank you for your help.

**Comment by dhoppenb** [comment]
Hi, 
While testing this I found an issue in the creation of mlops pipelines in scripts/create_ado_pipelines.sh line 26. Should be:
mlops_pipeline_files=$(ls $path_to_mlops_pipelines).
Other than that it works well :).

**Comment by dhangerkapil** [comment]
Two Issues
1. In initialise-project.yml change initialise_repo.sh to initalise_repo.sh or change the name of script to initialise_repo.sh
2. As mentioned above by @dhoppenb change the scripts/create_ado_pipelines.sh. Should be:
mlops_pipeline_files=$(ls $path_to_mlops_pipelines)
 

We need documentation for this. Some documentation will be helpful for the settings of devops pipelines and repos to run initialise-proj successfully. Happy to help with that. Will do more tests for infra and mlops pipelines next week.

File: .azuredevops/initialise-project.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  default: mlops-project-template
+
+- name: projectType
+  displayName: ML project type
+  type: string
+  default: classical
+  values:
+  - classical
+  - cv
+
```
Comment (cindyweng) [inline .azuredevops/initialise-project.yml:29]: need to add NLP here

File: .azuredevops/scripts/create_ado_pipelines.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+mlops_pipeline_files=$(ls $path_to_mlops_pipelines)
+
+for file in $mlops_pipeline_files
+do
+    az pipelines create \
+        --name ${file%.*} \
+        --detect true \
+        --description "Automatically created pipeline for MLOps $file" \
+        --repository $repo_name \
+        --branch here-is-your-template \
```
Comment (cindyweng) [inline .azuredevops/scripts/create_ado_pipelines.sh:29]: Can we make this branch naming "main"? What is this branch actually pointing to at the moment?

**Review by cindyweng — APPROVED** [review]

**Pull request merged**